src/AggregateFunctions/AggregateFunctionGroupBitmapData.h

```diff
@@ -579,6 +579,59 @@ class RoaringBitmapWithSmallSet : private boost::noncopyable
         }
     }
 
+    UInt64 rb_offset_limit(UInt64 offset, UInt64 limit, RoaringBitmapWithSmallSet & r1) const
+    {
+        if (limit == 0 || offset >= size())
+            return 0;
+
+        if (isSmall())
+        {
+            UInt64 offset_count = 0;
+            std::vector<T> answer;
+            for (const auto & x : small)
+            {
+                T val = x.getValue();
+                if (offset_count >= offset)
+                {
+                    answer.push_back(val);
+                }
+                else
+                {
+                    offset_count++;
+                }
+            }
+            if (limit < answer.size())
+            {
+                std::nth_element(answer.begin(), answer.begin() + limit, answer.end());
```

Comment:
This will give `limit` values which are the smallest in `[offset, end)`. Should it work like this?